### PR TITLE
[TTAHUB-3326] Fix some mailer bugs

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
- Digests were not correctly logging "onCompleteNotification"
- An invalid constant was used for one of the jobs
- TR notifications had start date and end date backwards

## How to test
- Confirm that digests send without error (this is onerous to set up locally, I apologize)
- Confirm that the correct constant was used the recipient report approved digest job
- Confirm that POC notifications will send for the TR report "missingSessionInfo" job

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3326

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/lib/mailer/index.js [hunk 1]
```diff
@@ -101,11 +101,18 @@ export const onFailedNotification = (job, error) => {
 };
 
 export const onCompletedNotification = (job, result) => {
-  if (result != null) {
-    logger.info(`Successfully sent ${job.name} notification for ${job.data.report.displayId || job.data.report.id}`);
-    logEmailNotification(job, true, result);
+  if (job.data.reports && Array.isArray(job.data.reports)) {
+    job.data.reports.forEach((report) => {
+      if (result != null) {
```
Comment (AdamAdHocTeam) [inline src/lib/mailer/index.js:106]: Looks like this assumes if we have a result then we will have a report.displayId? Should the if statment check anything that we are accessing here in the loop?

File: src/lib/mailer/index.js [hunk 2]
```diff
@@ -101,11 +101,18 @@ export const onFailedNotification = (job, error) => {
...
+        logger.info(`Successfully sent ${job.name} notification for ${report.displayId}`);
+      } else {
+        logger.info(`Did not send ${job.name} notification for ${report.displayId} preferences are not set or marked as "no-send"`);
+      }
+    });
+  } else if (result != null) {
```
Comment (AdamAdHocTeam) [inline src/lib/mailer/index.js:112]: Again is there any concern for accessing null values with things like job.data.report.displayId?

**Review by AdamAdHocTeam — APPROVED** [review]
Overall looks good and I was able to test most email cases.

Left minor suggestions on code restructure.

**Comment by thewatermethod** [comment]
image
Found by @kryswisnaskas during her review, although unrelated will try to address here

**Review by GarrettEHill — APPROVED** [review]

**Comment by kryswisnaskas** [comment]
I am still seeing an error if there is a failed notification (in `onFailedNotification`)

Following is the `job.data` that is being sent to the `logger` for the `reportApprovedDigest` (the name is obfuscated)

```
{
  user: {
    fullName: 'Joe Smith',
    nameWithNationalCenters: 'Joe Smith, ',
    id: 907,
    homeRegionId: 6,
    hsesUserId: '56181',
    hsesUsername: 'Joe Smith@redacted.com',
    hsesAuthorities: [ 'ROLE_FEDERAL' ],
    name: 'Joe Smith',
    phoneNumber: null,
    email: 'Joe Smith@redacted.com',
    flags: [],
    lastLogin: '2024-10-27T10:46:13.505Z',
    createdAt: '2024-09-13T16:43:51.904Z',
    updatedAt: '2024-10-27T10:46:13.505Z',
    validationStatus: [ [Object] ]
  },
  reports: [],
  type: 'reportApprovedDigest',
  freq: 'today',
  subjectFreq: 'daily'
}
line 100
2024-11-08T16:23:10.408Z AUDIT error:   Queue encountered an error: Cannot read properties of undefined (reading 'displayId') {}
```

**Comment by kryswisnaskas** [comment]
I verified that the digest notification logging is still outstanding. We have that covered in TTAHUB-2099 though.

**Comment by thewatermethod** [comment]
@kryswisnaskas I believe I've addressed your feedback

**Review by kryswisnaskas — APPROVED** [review]
Looks good! 👍

**Pull request merged**